### PR TITLE
Vinethud vote notification

### PR DESCRIPTION
**Pull request by Zevotech**
I hope you all will forgive my transgressions.

## About The Pull Request

Adds the Vine Thud as a usable emote. Does not come with a bind option, which is probably for the best.

## Why It's Good For The Game

Some people wanted it, and now its here. May Big Moth judge us all equally.

## Changelog
:cl:
add: Vine Thud Emote
/:cl:

### CODE REVIEW

**Comment by Sun-Soaked** [comment]
I'll accept this on the grounds that it can only be played during hardcrit, and instantly kills the user

**Comment by Zevotech** [comment]
> I'll accept this on the grounds that it can only be played during hardcrit, and instantly kills the user

We'll see how unfunny it gets after the first test, because while I totally agree with this idea, I have this tiny morsel of hope that this wont get murdered to death. A change that might come with the hardcrit only nerf would be replacing the round vote sound with the thud to get more attention towards voting (as suggested by neo42).

**Comment by Pockets-byte** [comment]
I wholly support this, absolutely nothing wrong with it at all nope

**Comment by Zevotech** [comment]
> I wholly support this, absolutely nothing wrong with it at all nope

That's the spirit! Let your poorly placed faith condemn my soul to coder hell forever!
:]

**Comment by HighPrincessErinys** [comment]
i remember when marg hacked together the ability to do this for one like one day or one round and it was gone after that


_and now its returned._

**Comment by TheNeoGamer42** [comment]
if you do make it a vote or ahelp sound, totally keep the emote, except its admin toggled, and doesn't display a chat message for others, so all of a sudden  an admin can just make any player who is aware just able to fucking troll the shit out of people and people just keep hearing *thud out of nowhere with no reason as to how or why

**Comment by terg-but-ius** [comment]
make the sound effect have a very low chance of triggering when someone does the *eyebrow emote

**Comment by Sun-Soaked** [comment]
> make the sound effect have a very low chance of triggering when someone does the *eyebrow emote

this actually, would be the best implementation of this

not effectively spammable as hard, unexpected and thus funny in the moment

**Comment by ghost** [comment]
god save us

**Comment by Zevotech** [comment]
yeah fuck it why not

**Comment by Zevotech** [comment]
i accidentally added the pull to here for the code that replaces the votesound with the vine thud so i may as well get this TM'ed anyways

**Comment by ValoTheValo** [comment]
we live in a society

File: code/modules/mob/emote.dm [hunk 1]
```diff
@@ -85,3 +85,17 @@
...
+	mob_type_allowed_typecache = list(/mob/living, /mob/dead/observer)
+	mob_type_ignore_stat_typecache = list(/mob/dead/observer)
+
+/datum/emote/thud/run_emote(mob/user, params,  type_override, intentional)
+	. = ..()
+	if(.)
+		if(isobserver(user) && prob(90)) //they know a little about rolling down in the deep
+			user.playsound_local(get_turf(user), 'sound/voice/vinethud.ogg', 50, TRUE)
+			return
+		playsound(user, 'sound/voice/vinethud.ogg', 50, 1, -1)
```
Comment (triplezeta) [inline code/modules/mob/emote.dm]: So, the vine thud is funny and all but in seriousness i don't think that *thud will be good for us. If you can remove the emote or make it an admin-only thing we can get this merged.

Comment (Zevotech) [reply code/modules/mob/emote.dm]: commented out the emote for later, at least until i learn how to make shit admin only lol. having no code is better than making shitcode.

**Comment by triplezeta** [comment]
should probably change the title of this, actually

File: code/modules/mob/emote.dm [hunk 2]
```diff
@@ -85,3 +85,17 @@
...
+	mob_type_allowed_typecache = list(/mob/living, /mob/dead/observer)
+	mob_type_ignore_stat_typecache = list(/mob/dead/observer)
+
+/datum/emote/thud/run_emote(mob/user, params,  type_override, intentional)
+	. = ..()
+	if(.)
+		if(isobserver(user) && prob(90)) //admin only until further notice :sadge: 
+			user.playsound_local(get_turf(user), 'sound/misc/vinethud.ogg', 50, TRUE)
+			return
+		playsound(user, 'sound/misc/vinethud.ogg', 50, 1, -1) */
```
Comment (Kapu1178) [inline code/modules/mob/emote.dm]:
```suggestion
```
Delete unused code.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Comment by triplezeta** [comment]
this pr can be merged as soon as the existing review is addressed

**Comment by MarkSuckerberg** [comment]
It's funny, but I really think we might need to find another sound that's more fitting, according to the handful of off-put reactions to it.

**Comment by MarkSuckerberg** [comment]
Merging on the condition that someone finds a better sound in the near future.

**Pull request merged**